File: clients/python/Bots/my_ai.py
from Bots.bot_ai import BotAI
from Bots.data import PlayState


class MyAI(BotAI):
    column = 0
    my_coin_id = -1

    def play(self, state: PlayState):
        if self.my_coin_id == -1:
            self.my_coin_id = state.coin_id
        # todo: put your logic code here
        return self.column

    def get_name(self):
        return self.name

    def __init__(self):
        # todo: give your bot a super duper cool name
        self.name = "MyAi"

File: clients/dart/bots/my_ai.dart
import '../play_state.dart';
import 'bot_ai.dart';

class MyAI implements BotAi {
  int _column = 0;
  int _myCoinId = -1;

  MyAI();

  @override
  int play(PlayState playState) {
    if (_myCoinId == -1 && playState.coinId != null) {
      _myCoinId = playState.coinId!;
    }
    //  todo: put your logic code here
    return _column;
  }

  @override
  String getName() {
    //  todo: give your bot a super duper cool name
    return "YourBotName";
  }
}